bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/servlets/PageTemplateRenderConditionServlet.java

```diff
@@ -34,6 +34,8 @@
 /**
  * This servlet handles <code>granite:rendercondition</code> requests from the CIF page component,
  * in order to decide if the product and category pickers should be displayed in the page properties dialog.
+ *
+ * @deprecated
```

Comment:
Why no remove this Servlet? This is "internal" used, nothing customers directly dependent on. And if you replace it we don't need it. Customers which don't upgrade can still use the older version of the bundle.

Reply:
In principle this server is public API because it's exposed publicly via the resource type core/cif/components/renderconditions/pagetemplate. If a customer is using it somewhere in a similar way as it was used in the archetype then that code will break. 
Strictly speaking the new servlet is not a drop-in replacement for this because it's doing different kind of checks and it's not using templates, it's part of a different approach to solve the issue in the page properties dialog.

Nevertheless, if we decide to remove it, I'm also fine with that because the likelihood of somebody using it elsewhere may be quite small. I just thought that deprecation is the safer way.

Reply:
> Nevertheless, if we decide to remove it, I'm also fine with that because the likelihood of somebody using it elsewhere may be quite small. I just thought that deprecation is the safer way.
+1 for remove, the number of customers using this might be overseeable. Most might be even on CIF versions before it was introduced.